### DISCUSSION TITLE
Basic question of how to use it

### DISCUSSION THREAD

**Discussion by anarqist** [abiosoft/colima#1155] [Q&A]
Hi, 
I am fairly new to the Docker world, so I hope my questions are not too basic and they may help othera as well. 

1. Is it possible to use Colima without Docker (Desktop) for Mac? If yes are there any downsides or issues if I would not have Docker installed too? 
2. Can I basically just start my already set up docker-compose files with Colima, or would it be better to install Portainer (if possible) on Colima and manage my Docker inside there?
3. The most important question why I even found Coloma: I have an issue adding a external drive connected to my Mac inside Docker to mount it and use it. If I would be able to add my volume to Colima, would it be then reachable within Container set up in "native" Docker/Portainer, or is ist restricted to the Colima environment? 

Last but not least, I would LOVE to ditch Docker for Colima, using OSS is my goal where ever I am and to try to support the community, so thanks for this project! 

Thanks in advance for any assistance and help it is mich appreciated!

**Comment by abiosoft** [comment]
Well, for the most part it should be a like for like substitute for Docker Desktop with the only obvious difference being the absence of a GUI desktop app, it runs in the background.

So I'd say drag Docker Desktop to Trash, and install Colima.

```sh
# install docker client and docker compose
brew install docker docker-compose
# install Colima
brew install colima
```

Then start it.

```
colima start
```
Lastly, go ahead and interract with Docker like you normally do and see if everything works.

**Reply by anarqist** [reply]
Thanks for the quick reply. 

I installed colima with homebrew and started it with the mentioned command. Then I was a bit confused but figured it out to use the "standard" docker commands to create a new one and I was able to deploy my first container with Colima 🥳

Portainer is up and running. I restarted colima and portainer was reset back to default. Portainers data is stored unter ~/folder/portainer. I understood the $HOME folder (and subfolders) are natively mounted and data persists a restart of colima? Or am I most likely something missing? 

I also added in the config file my external volume 

```
mounts:
  - location: /Volumes/NAMEOFMYVOLUME
    writable: true
```

I tried to use then a subfolder of my external volume to store data from Portainer into it to see if that even works and when I try to deploy my container I receive this message: 

`docker: Error response from daemon: error while creating mount source path '/Volumes/NAMEOFMYVOLUME/colima/portainer': chown /Volumes/NAMEOFMYVOLUME/colima/portainer: permission denied.` 

Where are my newbie mistakes here? 🥲

EDIT: I also tried to start colima with:

`colima start --mount=/Volumes/NAMEOFMYVOLUME`
but I still get the message in the logs: `2024/10/04 08:46PM FTL github.com/portainer/portainer/api/cmd/portainer/main.go:75 > failed creating file service | error="mkdir /data/certs: read-only file system"`

**Comment by abiosoft** [comment]
Yeah, once you specify custom mounts, you have overriden Colima's default mount config and would need to manually include $HOME in the list of mounts.

**Reply by anarqist** [reply]
Hi, 
Ah okay got it. So I need to add each path in the config file to mount them. 
is there then a hierarchy like if I first have my external drive but configs for container are stored on the internal storage, it will not survive a restart, or are they all handled the same? 
and do I need to add subfolders to in my mounted volumes or will everything be accessible recursive like ~/* or ~/folderA ~/folderA/folderB etc? 

And any explanation why I got the permissions error? The container was able to create subfolders just fine bit not the content inside the last folder, then I received the error. 

Thx in advance! ✌️
